static/js/ducks/source.js

```diff
@@ -77,25 +77,47 @@ export function addComment(formData) {
     return (dispatch) => {
       fileReaderPromise(formData.attachment).then((fileData) => {
         formData.attachment = fileData;
-        dispatch(API.POST(`/api/comment`, ADD_COMMENT, formData));
+
+        if (formData.spectrum_id) {
+          dispatch(
+            API.POST(
+              `/api/spectrum/${formData.spectrum_id}/comment`,
+              ADD_COMMENT,
+              formData
+            )
+          );
+        } else {
+          dispatch(
+            API.POST(
+              `/api/sources/${formData.obj_id}/comment`,
+              ADD_COMMENT,
+              formData
+            )
+          );
+        }
       });
     };
   }
-  return API.POST(`/api/comment`, ADD_COMMENT, formData);
+  if (formData.spectrum_id) {
+    return API.POST(
+      `/api/spectrum/${formData.spectrum_id}/comment`,
+      ADD_COMMENT,
+      formData
+    );
+  }
+  return API.POST(
+    `/api/sources/${formData.obj_id}/comment`,
+    ADD_COMMENT,
+    formData
+  );
 }
 
-export function deleteComment(comment_id, associatedResourceType = "object") {
-  return API.DELETE(
-    `/api/comment/${comment_id}/${associatedResourceType}`,
-    DELETE_COMMENT
-  );
+export function deleteComment(pathToComment) {
+  return API.DELETE(pathToComment, DELETE_COMMENT);
```

Comment:
I think we should break this up into two separate action creators for the different resource types; that makes more sense than for the full request path to be passed in as a parameter as is done here. That path should ideally just be hard-coded into each action creator body.

Reply:
This is mainly for internal use, and would improve readability, so seems okay to have one action per type.

Reply:
Exactly. Please update.